### ISSUE TITLE
[FEAT] Basic accessibility updates for DaffMenuComponent

### ISSUE THREAD

**Issue by xelaint** [graycoreio/daffodil#2841]
# :bulb: Feature request

## Feature Name
Accessibility updates for DaffMenuComponent

## The Desired Behavior

- [x] `aria-haspopup` and `aria-expanded` on DaffMenuComponent
- [x] Arrow key functionalities (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role)

## Your Use Case
As a developer, I need to provide an accessible menu component for users.

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by xelaint** [comment]
Currently blocked because `aria-controls` implementation doesn't work if using template ref to implement menu

**Comment by xelaint** [comment]
May no longer need to use templateRef.

**Comment by damienwebdev** [comment]
Closed by #3998
